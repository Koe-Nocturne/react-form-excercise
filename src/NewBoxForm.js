import React from 'react';



class NewBoxForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { backgroundColor: '', width: '', height: ''}
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }


  handleChange(evt) {
       this.setState({[evt.target.name]: evt.target.value})
  }
  handleSubmit(evt) {
    evt.preventDefault();
    this.props.onSubmit(this.state);
    this.setState({backgroundColor: '', width: '', height: ''});

  }

  render() {
  return (
    <div>
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="backgroundColor">Background color: </label>
        <input id="backgroundColor" name="backgroundColor" value={this.state.backgroundColor} onChange={this.handleChange}></input>

        <label htmlFor="width">Width: </label>
        <input id="width" name="width" value={this.state.width} onChange={this.handleChange}></input>

        <label htmlFor="height">Height: </label>
        <input id="height" name="height" value={this.state.height} onChange={this.handleChange}></input>
        <button>Create a new Box!</button>

      </form>
    </div>
  );
  }
}

export default NewBoxForm;