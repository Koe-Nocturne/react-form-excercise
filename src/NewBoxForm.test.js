import React from 'react';
import NewBoxForm from './NewBoxForm';
import toJson from "enzyme-to-json";
import { shallow, mount } from 'enzyme';


it("renders without crashing", () => {
  shallow(<NewBoxForm/>)
});

it("matches snapshot", () => {
  let wrapper = shallow(<NewBoxForm/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});



describe("allows for changes in background color, height and width", () => {
  it("can we edit background color", () => {
    let wrapper = mount(<NewBoxForm/>);
    const bgColor = wrapper.find("#backgroundColor");
    bgColor.instance().value = "blueViolet";
    bgColor.simulate("change");

    expect(wrapper.state().backgroundColor).toEqual("blueViolet");

  });

});