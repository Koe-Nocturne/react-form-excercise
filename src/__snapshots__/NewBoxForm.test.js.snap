// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="backgroundColor"
    >
      Background color: 
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      onChange={[Function]}
      value=""
    />
    <label
      htmlFor="width"
    >
      Width: 
    </label>
    <input
      id="width"
      name="width"
      onChange={[Function]}
      value=""
    />
    <label
      htmlFor="height"
    >
      Height: 
    </label>
    <input
      id="height"
      name="height"
      onChange={[Function]}
      value=""
    />
    <button>
      Create a new Box!
    </button>
  </form>
</div>
`;
